x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -557,7 +557,10 @@ private PhysicalOperation planHashJoin(HashJoinExec join, LocalExecutionPlannerC
     }
 
     private ExpressionEvaluator.Factory toEvaluator(Expression exp, Layout layout) {
-        return EvalMapper.toEvaluator(exp, layout);
+        var shardContexts = (physicalOperationProviders instanceof EsPhysicalOperationProviders)
+            ? ((EsPhysicalOperationProviders) physicalOperationProviders).shardContexts()
+            : null;
+        return EvalMapper.toEvaluator(exp, layout, shardContexts);
```

Comment:
I suppose I'd delegate this to `EsPhysicalOperationProviders` instead of `instanceof` it, but this is a proof of concept so it gets the job done.